doc/usage/restructuredtext/directives.rst

```diff
@@ -96,138 +96,128 @@ The ``toctree`` directive is the central element.
-   use the ``name`` option to provide an implicit target name that can be
-   referenced by using :rst:role:`ref`::
+      Add a caption of the toctree.
 
-      .. toctree::
-         :caption: Table of Contents
-         :name: mastertoc
+      .. versionadded:: 1.3
 
-         foo
+   .. rst:directive:option:: name: label
+      :type: text
 
-   As with :dudir:`most directives <common-options>`,
-   you can use the ``class`` option to assign `class attributes`_::
+      A label name that can be referenced using :rst:role:`ref`.
+      This is a :dudir:`common option <common-options>`.
 
-      .. toctree::
-         :class: custom-toc
+      .. versionadded:: 1.3
+
+   .. rst:directive:option:: class: class names
+      :type: a list of class names separated by spaces
 
-   .. _class attributes: https://docutils.sourceforge.io/docs/ref/doctree.html#classes
+      Assign `class attributes`_.
+      This is a :dudir:`common option <common-options>`.
 
-   If you want only the titles of documents in the tree to show up, not other
-   headings of the same level, you can use the ``titlesonly`` option::
+      .. _class attributes: https://docutils.sourceforge.io/docs/ref/doctree.html#classes
 
-      .. toctree::
-         :titlesonly:
+      .. versionadded:: 7.4
 
-         foo
-         bar
+   .. rst:directive:option:: numbered
 
-   You can use "globbing" in toctree directives, by giving the ``glob`` flag
-   option.  All entries are then matched against the list of available
-   documents, and matches are inserted into the list alphabetically.  Example::
+      If you want to have section numbers even in HTML output, add the ``:numbered:``
+      option to the *top-level* toctree.  For example::
 
-      .. toctree::
-         :glob:
+         .. toctree::
+            :numbered:
 
-         intro*
-         recipe/*
-         *
+            foo
+            bar
 
-   This includes first all documents whose names start with ``intro``, then all
-   documents in the ``recipe`` folder, then all remaining documents (except the
-   one containing the directive, of course.) [#]_
+      Numbering then starts at the heading of ``foo``.  Sub-toctrees are
+      automatically numbered (don't give the ``numbered`` flag to those).
 
-   The special entry name ``self`` stands for the document containing the
-   toctree directive.  This is useful if you want to generate a "sitemap" from
-   the toctree.
+      Numbering up to a specific depth is also possible, by giving the depth as a
+      numeric argument to ``numbered``.
 
-   You can use the ``reversed`` flag option to reverse the order of the entries
-   in the list. This can be useful when using the ``glob`` flag option to
-   reverse the ordering of the files.  Example::
+      .. versionadded:: 0.6
 
-      .. toctree::
-         :glob:
-         :reversed:
+      .. versionchanged:: 1.1
+         Added numeric argument to "numbered".
 
-         recipe/*
+   .. rst:directive:option:: titlesonly
```

Comment:
Isn't this incorrect syntax? Why write it in a single line?

```rst
.. rst:directive::
   :option:
```

or 

```rst
.. toctree::
   :titlesonly:
```

Reply:
I've taken this from https://github.com/sphinx-doc/sphinx/blob/96b511798da0ab819b87be363fe505186896fee2/doc/usage/restructuredtext/directives.rst?plain=1#L665 and similar.

it's a bit peculiar naming, but `directive:option` is a directive in the`rst` domain:
https://github.com/sphinx-doc/sphinx/blob/96b511798da0ab819b87be363fe505186896fee2/sphinx/domains/rst.py#L229

Reply:
Ok, makes sense now :)